src/transformers/convert_slow_tokenizer.py

```diff
@@ -1504,6 +1505,75 @@ def converted(self) -> Tokenizer:
         return tokenizer
 
 
+class TekkenConverter:
+    """
+    A general converter used for mistral's `tekken.json`. The underlying is a tiktoken model.
+    """
+
+    def __init__(
+        self,
+        vocab_file=None,
+        pattern=r"""(?i:'s|'t|'re|'ve|'m|'ll|'d)|[^\r\n\p{L}\p{N}]?\p{L}+|\p{N}{1,3}| ?[^\s\p{L}\p{N}]+[\r\n]*|\s*[\r\n]+|\s+(?!\S)|\s+""",
+        add_prefix_space=False,
+        additional_special_tokens=None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args)
+        self.vocab_file = vocab_file
+        self.pattern = pattern
+        self.add_prefix_space = add_prefix_space
+        self.additional_special_tokens = additional_special_tokens
+
+    def extract_vocab_merges_from_model(self, tekken_url: str):
+        with open(tekken_url, "r") as f:
+            bpe_ranks = json.load(f)
```

Comment:
```suggestion
    with open(tekken_url, "r") as f:
        tekken_vocab = json.load(f)["vocab"]

    bpe_ranks = {}
    for item in tekken_vocab:
        rank, token_bytes = item["rank"], item["token_bytes"] # token_str ignored
        token = base64.b64decode(token_bytes)
        bpe_ranks[token] = rank
```

^ bpe_ranks can be obtained from their json file like this